src/services/chain.ts

```diff
@@ -228,67 +227,57 @@ export class ChainContext {
           log.debug(`Found ${events.length} events`);
         }
 
-        // process the events
-        for (const event of events) {
-          if (plan[event.blockNumber] === undefined) {
-            plan[event.blockNumber] = new Set();
+        // Get relevant block numbers to process (the ones with relevant events)
+        const eventsByBlock = events.reduce<
+          Record<number, ConditionalOrderCreatedEvent[]>
+        >((acc, event) => {
+          const events = acc[event.blockNumber];
+          if (events) {
+            events.push(event);
+          } else {
+            acc[event.blockNumber] = [event];
           }
 
-          plan[event.blockNumber].add(event);
+          return acc;
+        }, {});
+
+        // Process blocks in order
+        for (const blockNumberKey of Object.keys(eventsByBlock).sort()) {
+          const blockNumber = Number(blockNumberKey);
+          lastProcessedBlock = await processBlockAndPersist({
```

Comment:
Here `lastProcessedBlock` is set.
However, I don't see it used anywhere before it's overwritten on line `258`.

Reply:
Yes, you are right. I deleted it 858c00fd8006557182d84f3824c17d6d22c5dbf5

I think is just because during the warmup we search for events for a block range, once we find an "interesting block" we update the DB and prometheus metrics to signal that that's the last processed block. So that variable will always have the latest indexed block (even if we don't use it)

I could add a log entry instead of just deleting, but anyways, I decided to kill the assinment.